apps/dashboard/app/(app)/ratelimits/[namespaceId]/settings/components/settings-client.tsx

```diff
@@ -11,59 +12,51 @@ type Props = {
 };
 
 export const SettingsClient = ({ namespaceId }: Props) => {
-  const trpcUtils = trpc.useUtils();
   const [isNamespaceNameDeleteModalOpen, setIsNamespaceNameDeleteModalOpen] = useState(false);
-  const [isUpdating, setIsUpdating] = useState(false);
 
-  const { data, isLoading } = trpc.ratelimit.namespace.queryDetails.useQuery({
-    namespaceId,
-    includeOverrides: false,
-  });
+  const { data } = useLiveQuery((q) =>
+    q
+      .from({ namespace: collection.ratelimitNamespaces })
+      .where(({ namespace }) => eq(namespace.id, namespaceId)),
+  );
 
-  const [namespaceName, setNamespaceName] = useState(data?.namespace.name || "");
+  const namespace = data.at(0);
 
+  const [namespaceName, setNamespaceName] = useState<string | null>(null);
   useEffect(() => {
-    if (data?.namespace.name && namespaceName === "") {
-      setNamespaceName(data.namespace.name);
+    if (namespaceName === null && namespace) {
+      setNamespaceName(namespace.name);
     }
-  }, [data?.namespace.name, namespaceName]);
-
-  const updateNameMutation = trpc.ratelimit.namespace.update.name.useMutation({
-    onSuccess() {
-      toast.success("Your namespace name has been renamed!");
-      trpcUtils.ratelimit.namespace.query.invalidate();
-      trpcUtils.ratelimit.namespace.queryDetails.invalidate();
-      setIsUpdating(false);
-    },
-    onError(err) {
-      toast.error("Failed to update namespace name", {
-        description: err.message,
-      });
-      setIsUpdating(false);
-    },
-  });
+  }, [namespace, namespaceName]);
 
   const handleUpdateName = async () => {
-    if (!data?.namespace) {
+    if (!namespace) {
       return;
     }
 
-    if (namespaceName === data.namespace.name || !namespaceName) {
+    if (namespaceName === namespace.name || !namespaceName) {
       return toast.error("Please provide a different name before saving.");
     }
+    let error = "";
+    collection.ratelimitNamespaces.forEach((ns) => {
+      if (ns.id !== namespaceId && namespaceName === ns.name) {
+        error = "Another namespace already has this name";
+        return;
+      }
+    });
+    if (error) {
+      return toast.error(error);
+    }
 
-    setIsUpdating(true);
-    await updateNameMutation.mutateAsync({
-      name: namespaceName,
-      namespaceId: data.namespace.id,
+    collection.ratelimitNamespaces.update(namespace.id, (draft) => {
+      draft.name = namespaceName;
     });
   };
 
-  if (!data || isLoading) {
+  if (!namespace) {
```

Comment:
Infinity skeleton if you land somewhere and namespace doesn't exist, is this what you wanted?

Reply:
fixed
<img width="2350" height="898" alt="CleanShot 2025-09-10 at 08 41 35@2x" src="https://github.com/user-attachments/assets/fe1515c7-4779-45e4-b22f-c412ac5e04a3" />